src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -586,10 +597,41 @@ LRESULT CALLBACK FancyZones::s_WndProc(HWND window, UINT message, WPARAM wparam,
         DefWindowProc(window, message, wparam, lparam);
 }
 
-void FancyZones::UpdateZoneWindows() noexcept
+bool FancyZones::IsWindowInteresting(HWND hwnd)
 {
-    auto callback = [](HMONITOR monitor, HDC, RECT *, LPARAM data) -> BOOL
+    // Get general window ifo
+    auto windowInfo = get_filtered_window_info(hwnd);

```

Comment:
Before `get_filtered_window` would just say if the window should be filtered and return the path to the executable. Now it provides some more information. Some of the windows (Start menu, taskbar, buttons) are filtered like before, but for the rest of the windows, we also get information if the window has an owner and if it is a "standard" window (none of the `WS_DISABLED`, `WS_EX_TOOLWINDOW` nor `WS_EX_NOACTIVATE` styles set).

Normally, we return true only for standard windows without an owner. But if the app is in the approved apps list, we also return true here.